docker/local/Dockerfile

```diff
@@ -0,0 +1,24 @@
+FROM golang:1.18-alpine AS builder
+
+WORKDIR /polygon-edge
+
+ADD go.mod go.sum ./
+RUN go mod download
+
+COPY . .
```

Comment:
Can we also add in a `.dockerignore` to prevent potential binary inclusions during this build process? Helps reduce the room for human errors down the line. :pray:

Reply:
Done

Reply:
To clarify, I was referring to preventing possible pre-built binaries for any users testing locally as a result of `go build` or `make build`, since those will be overwritten/unused.

Reply:
if you are talking about `polygon-edge` binary, it will be overwritten in the next step.

Reply:
Yes, it will, but its unnecessary extra entropy to prebloat a cache layer. This was was more of a nitpick tbh. :+1: